packages/dev/core/src/FlowGraph/flowGraphEventCoordinator.ts

```diff
@@ -0,0 +1,28 @@
+import { Observable } from "../Misc/observable";
+import type { FlowGraphCustomEvent } from "./flowGraphCustomEvent";
+
+/**
+ * This class centralizes the handling of events in the flow graph.
+ * If a block fires an event, it will be handled by this class.
+ * It will also notify any blocks that are listening to that event.
+ * @experimental
+ */
+export class FlowGraphEventCoordinator {

```

Comment:
General thought - this is a global coordinator. Would it make sense to have internal coordinator per context, or do we support cross-context events? Would it make sense to provide an optional context, which keeps the scope of the event coordinator within this specific context? This way a dev could raise the same type of event, but when providing context it will only be executed in blocks of that context, providing a more granular control of events.

But again - just a thought, I am not even sure it is fully possible.

Reply:
For the GLTF spec, we want the events to always be cross context, since their purpose is to communicate between graphs. But we could add context specific events in the future too, if that is useful :)